search/search_repository.go

```diff
@@ -0,0 +1,357 @@
+package search
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+
+	"log"
+	"strconv"
+
+	"strings"
+
+	"regexp"
+
+	"net/url"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/models"
+	"github.com/asaskevich/govalidator"
+	"github.com/jinzhu/gorm"
+)
+
+const (
+	/*
+		- The SQL queries do a case-insensitive search.
+		- English words are normalized during search which means words like qualifying === qualify
+		- To disable the above normalization change "to_tsquery('english',$1)" to "to_tsquery($1)"
+		- Create GIN indexes : https://www.postgresql.org/docs/9.5/static/textsearch-tables.html#TEXTSEARCH-TABLES-INDEX
+		- To perform "LIKE" query we are appending ":*" to the search token
+
+	*/
+
+	// This SQL query is used when search is performed across workitem fields and workitem ID
+	WhereClauseForSearchByText = `setweight(to_tsvector('english',fields->>'system.title'),'B')||
```

Comment:
@sbose78 : IMO, we shold `somehow` mention that this `where clause` is only for `workitem` relation ?
We can maintain a constants in a table like -


   | search by ID | full text search query
------------ |------------ | -------------
work_item | WHERE_CLAUSE_WI_BY_ID | WHERE_CLAUSE_WI_BY_FULL_TEXT
users | WHERE_CLAUSE_USER_BY_ID | WHERE_CLAUSE_USER_BY_FULL_TEXT


Because soon we will have `type` in query param that will decide which table to query on.
We can take `union` of multiple query results in the DB itself.